src/linear_solver_routines/randomized_arnoldi_solver.jl

```diff
@@ -0,0 +1,98 @@
+# Date: 09/04/2024
+# Author: Christian Varner
+# Purpose: Implement the randomized arnoldi solver from
+# "Randomized Orthogonal Projection Methods for Krylov Subspace Solvers", 
+# by Timsit, Grigori, Balabanov.
+
+"""
+    randomized_arnoldi!(x, A, b, k; sketch_matrix)
+
+Find an approximate solution of the linear system using the randomized arnoldi iteration.
+This method was introduced in the following paper
+
+Timsit, Grigori, and Balabanov. "Randomized Orthogonal Projection Methods for Krylov Subspace Solvers."
+https://arxiv.org/abs/2302.07466.
+
+# Parameters
+- `x::AbstractVector`, initial solution to the linear system `Ax = b`. Will be overwritten.
+- `A::AbstractMatrix`, coefficient matrix for linear system.
+- `b::AbstractVector`, constant vector for linear system.
+- `k::Int`, number of iterations to do (size of Krylov Subspace).
+- `sketch_matrix::Union{AbstractMatrix, Nothing}` (Optional; Default value `nothing`) 
+"""
+function randomized_arnoldi!(
+    x::AbstractVector,                                      # initial guess - will be overwritten
+    A::AbstractMatrix,                                      # coefficient matrix
+    b::AbstractVector,                                      # constant vector
+    k::Int;                                                 # max iteration limit 
+    sketch_matrix::Union{AbstractMatrix, Nothing} = nothing # optional sketch matrix
+)
+
+    # helper function to set up and solve linear system after main iteration
+    function solve_linear_system!(
+        x::AbstractVector, 
+        beta::Float64, 
+        H::AbstractMatrix, 
+        V::AbstractMatrix, 
+        k::Int64)
+
+        ek = zeros(k)
+        ek[1] = beta
+        rho = H[1:k, 1:k] \ ek
+        x .+= V[:, 1:k] * rho
+    end
```

Comment:
Should be outside of the function and be given a descriptive name. Can put in utility or helper folder. Confirm with @nathanielpritchard